lib/cudadrv/context.jl

```diff
@@ -311,13 +311,16 @@ function CuDevice(ctx::CuContext)
 end
 
 """
-    device_synchronize()
+    synchronize(ctx::Context)
 
-Block for the current device's tasks to complete. This is a heavyweight operation, typically
-you only need to call [`synchronize`](@ref) which only synchronizes the stream associated
-with the current task.
+Synchronize the current context, waiting for all outstanding operations to complete.
+
+!!! warning
+
+    This is an operation that blocks in the driver, and should be avoided if possible.
+    Instead, use [`device_synchronize()`](@ref) to perform synchronization in Julia.
 """
-function device_synchronize()
+function synchronize(ctx::CuContext)
     cuCtxSynchronize()
```

Comment:
Should `ctx` be passed to `cuCtxSynchronize`?

Reply:
Oh oops, we need to switch contexts first (sadly this API call doesn't take the context...).